### PR TITLE
fix: Don't throw exception when getting representation of unrun GCA objects

### PR DESCRIPTION
**Pull request by samgoodman**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [X] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-aiplatform/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [X] Ensure the tests and linter pass
- [X] Code coverage does not decrease (if any source code was changed)
- [] Appropriate docs were updated (if necessary)

Fixes b/197264919 🦕

### CODE REVIEW

File: tests/unit/aiplatform/test_custom_job.py [hunk 1]
```diff
@@ -661,3 +661,36 @@ def test_create_from_local_script_with_all_args(
         assert (
             job._gca_resource.state == gca_job_state_compat.JobState.JOB_STATE_SUCCEEDED
         )
+
+    def test_check_custom_job_availability(self, get_custom_job_mock, create_custom_job_mock):
```
Comment (sasha-gitg) [inline tests/unit/aiplatform/test_custom_job.py:665]: We're moving towards using `@pytest.mark.usefixtures` when we don't consume the fixture in the test directly (asserting some state against it) in the test:

```suggestion
    @pytest.mark.usefixtures('get_custom_job_mock', 'create_custom_job_mock')
    def test_check_custom_job_availability(self):
```

**Review by sasha-gitg — COMMENTED** [review]
LGTM! Minor comments.

File: google/cloud/aiplatform/base.py [hunk 2]
```diff
@@ -655,6 +655,15 @@ def gca_resource(self) -> proto.Message:
         self._assert_gca_resource_is_available()
         return self._gca_resource
 
+    @property
+    def resource_is_available(self) -> bool:
```
[the remark this answers is not in the document]
Comment (samgoodman) [reply google/cloud/aiplatform/base.py]: Done

File: tests/unit/aiplatform/test_custom_job.py (hunk 1 above)
Comment (samgoodman) [reply tests/unit/aiplatform/test_custom_job.py]: Done

**Review by sasha-gitg — APPROVED** [review]

**Pull request merged**